File: in_Python/0219 Contains Duplicate II.py
class Solution(object):
    def containsNearbyDuplicate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: bool
        """
        dict = {}
        for i in range(len(nums)):
            if nums[i] not in dict:
                dict[nums[i]] = i
            else:
                if i - dict[nums[i]] <= k:
                    return True
                dict[nums[i]] = i
        return False

File: in_Java/0219 Contains Duplicate II.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        Map<Integer,Integer> hm = new HashMap<>();
        for(int i = 0; i < nums.length; i++){
            int cur = nums[i];
            if(!hm.containsKey(cur)){
                hm.put(cur,i);
            }else{
                if(Math.abs(i - hm.get(cur)) <= k){
                    return true;
                }else{
                    hm.put(cur,i);
                }
            }
        }
        return false;
    }
}